mac80211: update to v6.14.5

namiltd opened the pull request:
Update Backports mac80211 to version 6.14.5.

Tested successfully on routers:

- ramips/mt7620 - TP-Link Archer C5 v4 (kernel 6.6):
```
[   20.757735] Loading modules backported from Linux version v6.14.5-0-gaeaee199900e
[   20.765475] Backport generated by backports.git v6.1.110-1-43-g2e32643b84e8
[   21.617590] mt76x2e 0000:01:00.0: card - bus=0x1, slot = 0x0 irq=4
[   21.624222] mt76x2e 0000:01:00.0: ASIC revision: 76120044
[   22.317413] mt76x2e 0000:01:00.0: ROM patch build: 20141115060606a
[   22.349855] mt76x2e 0000:01:00.0: Firmware Version: 0.0.00
[   22.355583] mt76x2e 0000:01:00.0: Build: 1
[   22.359766] mt76x2e 0000:01:00.0: Build Time: 201607111443____
[   22.425305] mt76x2e 0000:01:00.0: Firmware running!
[   22.430507] mt76x2e 0000:01:00.0: registering led 'mt76-phy0'
```
- ramips/mt7620 - Xiaomi Mi Router 3 (kernel 6.6)
- ramips/mt7621 - Xiaomi Mi Router 4A Gigabit Edition (kernel 6.6)
- ramips/mt7621 - Beeline Smartbox pro, Wifire NBN-S1500 (kernel 6.12)
- ramips/mt7621 - Xiaomi Mi router 4 GE (kernel 6.12)
- mediatek/mt7622 - TOTOLINK A8000RU (kernel 6.12)
- mediatek/mt7622 - Xiaomi AX6s (kernel 6.12)
- mediatek/mt7986a - Zyxel EX5601-T0 (kernel 6.12)
- mediatek/mt7986a - ASUS TUF-AX4200 (kernel 6.12)
- mediatek/filogic - Bananapi BPi-R4
- mediatek/filogic  - Cudy TR3000 (kernel 6.12)

Tested successfully on USB WiFi adapters:
- mediatek/mt7601u - x86_64, VirtualBox (kernel 6.12):
```
[   17.768140] Loading modules backported from Linux version v6.14.5-0-gaeaee199900e
[   17.786237] Backport generated by backports.git v6.1.110-1-43-g2e32643b84e8
[   17.816138] e1000e: Intel(R) PRO/1000 Network Driver
[   17.830031] e1000e: Copyright(c) 1999 - 2015 Intel Corporation.
[   18.162710] usb 2-1: reset high-speed USB device number 2 using ehci-pci
[   18.521627] mt7601u 2-1:1.0: ASIC revision: 76010001 MAC revision: 76010500
[   19.352419] mt7601u 2-1:1.0: EEPROM ver:0d fae:00
[   20.923930] ieee80211 phy0: Selected rate control algorithm 'minstrel_ht'
[   20.924268] usbcore: registered new interface driver mt7601u
```
- realtek/rtw88_8821cu
- realtek/rtw88_8812au

No performance changes were found for kernel 6.6.92 and 6.12.31.

namiltd committed:
mac80211: update to version 6.14.5

Update to version 6.14.5

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd commented:
@Noltari, @mantas-p, @neheb I invite You to test it and continue. I'm stuck on fixes for ath12k and rtl.

namiltd committed:
mac80211: restore old debugfs_fops

Restore old debugfs_fops for compatibility with kernel 6.6 and 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

neheb commented:
opinion on https://github.com/neheb/openwrt/commit/8e906f961ac5693dac1ed0f0404e3a7cb6bccdd9 and https://github.com/neheb/openwrt/commit/a655f49f97d6139dcab103b1368b0bd3abf97bec ?

namiltd commented:
@naheb Unfortunately, I have no knowledge about the validity of removing these patches.

rmandrad commented:
tested on mediatek/filogic on kernel 6.12 by @rmandrad 

not related to this commit but mt76 (perhaps is kernel 6.12...) had to create two patches ... ping @nbd168 

1. 
`--- a/trace.h	2025-05-26 11:53:51.608046451 +0100
+++ b/trace.h	2025-05-26 11:53:43.898305358 +0100
@@ -14,8 +14,12 @@
 
 #define MAXNAME		32
 #define DEV_ENTRY	__array(char, wiphy_name, 32)
-#define DEVICE_ASSIGN	strlcpy(__entry->wiphy_name,	\
-				wiphy_name(dev->hw->wiphy), MAXNAME)
+#define DEVICE_ASSIGN                                         \
+    do {                                                      \
+        strncpy(__entry->wiphy_name,                          \
+                wiphy_name(dev->hw->wiphy), MAXNAME - 1);     \
+        __entry->wiphy_name[MAXNAME - 1] = '\0';              \
+    } while (0)
 #define DEV_PR_FMT	"%s"
 #define DEV_PR_ARG	__entry->wiphy_name
`

2.
`--- a/mt7996/main.c	2025-04-11 10:07:15.000000000 +0100
+++ b/mt7996/main.c	2025-05-26 13:25:00.572516722 +0100
@@ -664,7 +664,8 @@
 }
 
 static int
-mt7996_get_txpower(struct ieee80211_hw *hw, struct ieee80211_vif *vif, int *dbm)
+mt7996_get_txpower(struct ieee80211_hw *hw, struct ieee80211_vif *vif, 
+		unsigned int link_id, int *dbm)
 {
 	struct mt7996_vif *mvif = (struct mt7996_vif *)vif->drv_priv;
 	struct mt7996_phy *phy = mt7996_vif_link_phy(&mvif->deflink);
`

rmandrad commented:
posted the above too early ... changed the ssid and got a crash pstore shows the following

`Oops#1 Part1
<1>[  477.678817] [0000000000000008] pgd=0000000000000000, p4d=0000000000000000, pud=0000000000000000
<0>[  477.687516] Internal error: Oops: 0000000096000045 [#1] SMP
<7>[  477.693078] Modules linked in: pppoe ppp_async wireguard sch_mqprio sch_fq_pie pppox ppp_generic nft_fib_inet nf_flow_table_inet mt7996e(O) mt76_connac_lib(O) mt76(O) mac80211(O) libchacha20poly1305 chacha_neon cfg80211(O) act_sample xt_u32 xt_time xt_tcpmss xt_string xt_statistic xt_state xt_socket xt_recent xt_quota2(O) xt_quota xt_psd(O) xt_pkttype xt_physdev xt_owner xt_nat xt_multiport xt_mark xt_mac xt_lscan(O) xt_limit xt_length2(O) xt_length xt_ipv4options(O) xt_iprange xt_ipp2p(O) xt_iface(O) xt_hl xt_helper xt_hashlimit xt_geoip(O) xt_fuzzy(O) xt_esp xt_ecn xt_dscp xt_conntrack xt_connmark xt_connlimit xt_connlabel xt_connbytes xt_condition(O) xt_comment xt_cluster xt_cgroup xt_bpf xt_asn(O) xt_addrtype xt_TRACE xt_TPROXY xt_TEE xt_TCPMSS xt_SYSRQ(O) xt_REDIRECT xt_PROTO(O) xt_NFQUEUE xt_NFLOG xt_NETMAP xt_MASQUERADE xt_LUA(O) xt_LOGMARK(O) xt_LOG xt_LED xt_IPMARK(O) xt_HL xt_FLOWOFFLOAD xt_DSCP xt_DNETMAP(O) xt_DHCPMAC(O) xt_CT xt_CLASSIFY xt_CHECKSUM xt_DELUDE(O) xt_TARPIT(O) ipt_REJECT xt_tcpudp
<7>[  477.693260]  xt_CHAOS(O) xt_ACCOUNT(O) ums_usbat ums_sddr55 ums_sddr09 ums_karma ums_jumpshot ums_isd200 ums_freecom ums_datafab ums_cypress ums_alauda ts_fsm ts_bm tls tcp_scalable tcp_hybla tcp_bbr slhc sfp sch_skbprio sch_red sch_prio sch_pie sch_mqprio_lib sch_drr sch_cake rtc_pcf8563 psample poly1305_neon nft_tproxy nft_socket nft_reject_ipv6 nft_reject_ipv4 nft_reject_inet nft_reject_bridge nft_reject nft_redir nft_quota nft_queue nft_numgen nft_nat nft_meta_bridge nft_masq nft_log nft_limit nft_hash nft_fwd_netdev nft_flow_offload nft_fib_ipv6 nft_fib_ipv4 nft_fib nft_dup_netdev nft_dup_ipv6 nft_dup_ipv4 nft_ct nft_connlimit nft_compat nft_chain_nat nfnetlink_queue nfnetlink_log nfnetlink_cttimeout nfnetlink_cthelper nf_tproxy_ipv6 nf_tproxy_ipv4 nf_tables nf_socket_ipv6 nf_socket_ipv4 nf_reject_ipv4 nf_nat_tftp nf_nat_snmp_basic nf_nat_sip nf_nat_pptp nf_nat_irc nf_nat_h323 nf_nat_ftp nf_nat_amanda nf_log_syslog nf_flow_table nf_dup_netdev nf_dup_ipv6 nf_dup_ipv4 nf_conntrack_tftp nf_conntrack_snmp
<7>[  477.782609]  nf_conntrack_sip nf_conntrack_sane nf_conntrack_pptp nf_conntrack_netlink nf_conntrack_netbios_ns nf_conntrack_irc nf_conntrack_h323 nf_conntrack_ftp nf_conntrack_broadcast nf_conntrack_bridge ts_kmp nf_conntrack_amanda nf_conncount mptcp_diag mhi_pci_generic mhi mdio_netlink(O) mdio_i2c libcurve25519_generic libchacha iptable_raw iptable_nat iptable_mangle iptable_filter ipt_rpfilter ipt_ah ipt_ECN ip6table_raw ip6t_rpfilter ip_tables compat_xtables(O) compat(O) cls_flower br_netfilter at24 act_vlan act_police crypto_safexcel sch_teql sch_sfq sch_multiq sch_gred sch_fq sch_ets sch_codel em_text em_nbyte em_meta em_cmp act_skbmod act_simple act_pedit act_csum libcrc32c cls_bpf act_bpf act_ctinfo act_connmark sch_tbf sch_ingress sch_htb sch_hfsc em_u32 cls_u32 cls_route cls_matchall cls_fw cls_flow cls_basic act_skbedit act_mirred act_gact pwm_fan i2c_mux_pca954x i2c_mux cryptodev(O) xt_set ip_set_list_set ip_set_hash_netportnet ip_set_hash_netport ip_set_hash_netnet ip_set_hash_netiface ip_set_hash_net
<7>[  477.871785]  ip_set_hash_mac ip_set_hash_ipportnet ip_set_hash_ipportip ip_set_hash_ipport ip_set_hash_ipmark ip_set_hash_ipmac ip_set_hash_ip ip_set_bitmap_port ip_set_bitmap_ipmac ip_set_bitmap_ip ip_set nfnetlink ip6table_nat nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 ip6t_NPT ip6t_rt ip6t_mh ip6t_ipv6header ip6t_hbh ip6t_frag ip6t_eui64 ip6t_ah ip6table_mangle ip6table_filter ip6_tables ip6t_REJECT x_tables nf_reject_ipv6 ip6_gre ip_gre gre ifb ip6_tunnel netlink_diag tunnel6 ip_tunnel udp_diag tcp_diag raw_diag inet_diag veth tun autofs4 vxlan udp_tunnel ip6_udp_tunnel macsec rsa_generic mpi asn1_decoder akcipher crypto_user algif_skcipher algif_rng algif_hash algif_aead af_alg sha512_arm64 sha1_ce sha1_generic seqiv md5 geniv fcrypt des_generic libdes cts chacha20poly1305 cbc blake2b_generic authencesn authenc usb_storage leds_gpio xhci_plat_hcd xhci_pci xhci_mtk_hcd xhci_hcd ehci_pci fsl_mph_dr_of ehci_platform ehci_fsl ehci_hcd nvme nvme_core gpio_button_hotplug(O) usbcore usb_common aquantia crc_itu_t
<7>[  478.051729] CPU: 2 UID: 101 PID: 2532 Comm: hostapd Tainted: G           O       6.12.30 #0
<7>[  478.060073] Tainted: [O]=OOT_MODULE
<7>[  478.063551] Hardware name: Banana Pi BPI-R4 (2x SFP+) (DT)
<7>[  478.069024] pstate: 00400005 (nzcv daif +PAN -UAO -TCO -DIT -SSBS BTYPE=--)
<7>[  478.075975] pc : cfg80211_chandef_valid+0x664/0x70c [cfg80211]
<7>[  478.081826] lr : cfg80211_update_last_available+0x4c/0xa0 [cfg80211]
<7>[  478.088182] sp : ffffffc081efb7b0
<7>[  478.091486] x29: ffffffc081efb7b0 x28: ffffffc081efbdc8 x27: 0000000000000000
<7>[  478.098615] x26: ffffff80cdee03c0 x25: ffffffc079cb59c0 x24: 0000000000000000
<7>[  478.105741] x23: ffffff80c019c000 x22: 00000000000003e8 x21: ffffff80cdee03c0
<7>[  478.112867] x20: ffffff80cdee03c0 x19: 000000006be2e840 x18: 0000000000000000
<7>[  478.119992] x17: 0000000000000000 x16: 0000000000000000 x15: 0000007fa6195160
<7>[  478.127118] x14: 0000000000000000 x13: 0000000000000000 x12: 0000000000000000
<7>[  478.134243] x11: 0000000000000040 x10: 0000000000000002 x9 : 0000000000000000
<7>[  478.141368] x8 : ffffff80ce376a88 x7 : 0000000000000000 x6 : 000000000000003f
<7>[  478.148494] x5 : 0000000000000004 x4 : 0000000000000000 x3 : 00000000000003e8
<7>[  478.155619] x2 : 0000000000000140 x1 : 00000000005d27a8 x0 : ffffff80cdee03c0
<7>[  478.162745] Call trace:
<7>[  478.165180]  cfg80211_chandef_valid+0x664/0x70c [cfg80211]
<7>[  478.170671]  cfg80211_update_last_available+0x4c/0xa0 [cfg80211]
<7>[  478.176678]  cfg80211_leave_mesh+0x26c/0x430 [cfg80211]
<7>[  478.181905]  cfg80211_stop_ap+0xa8/0xc0 [cfg80211]
<7>[  478.186697]  nl80211_send_chandef+0xc00/0xd80 [cfg80211]
<7>[  478.192009]  genl_family_rcv_msg_doit+0xac/0x114
<7>[  478.196619]  genl_rcv_msg+0x1b0/0x244
<7>[  478.200271]  netlink_rcv_skb+0x5c/0x128
<7>[  478.204098]  genl_rcv+0x34/0x48
<7>[  478.207230]  netlink_unicast+0x1e8/0x2d4
<7>[  478.211142]  netlink_sendmsg+0x170/0x37c
<7>[  478.215054]  ____sys_sendmsg+0xb4/0x2a8
<7>[  478.218883]  ___sys_sendmsg+0x7c/0xc0
<7>[  478.222535]  __sys_sendmsg+0x44/0x9c
<7>[  478.226103]  __arm64_sys_sendmsg+0x20/0x28
<7>[  478.230189]  invoke_syscall.constprop.0+0x4c/0xd0
<7>[  478.234886]  do_el0_svc+0x3c/0xd0
<7>[  478.238191]  el0_svc+0x24/0xc4
<7>[  478.241237]  el0t_64_sync_handler+0x118/0x124
<7>[  478.245583]  el0t_64_sync+0x164/0x168
<0>[  478.249238] Code: 1b037c33 a9025bf5 aa0003f5 f9001bf7 (29010482) 
<4>[  478.255319] ---[ end trace 0000000000000000 ]---
`

namiltd commented:
@rmandrad It is easier to replace strlcpy with strscpy.

namiltd committed:
mt76: update for compatibility with backports 6.14.5

Update for compatibility with backports 6.14.5.
Adding link_id to get_txpower functions parameters
via 003-wifi-mt76-link_id.patch.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

rmandrad commented:
@namiltd I would suggest not to include the mt76 patches as part of this commit. The patches I created was just to to over the hurdle of having mt76 working with your changes. The mt76 patches should come from upstream that @nbd168 manages

namiltd commented:
I added the 398db](https://github.com/openwrt/openwrt/pull/18911/commits/44398db4945a4d3846d2480c8e3108fffbfa6a6f) patc at the same time as a separate PR: https://github.com/openwrt/openwrt/pull/18927

namiltd committed:
mac80211: use remove_new

Easy compability fix for kernel 6.6, 6.12 and above.
Patches merged into one file for all devices.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
mac80211: ti: change read_new to read

Change function read_new to read for compatibility
with kernel 6.6 and 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

mantas-p committed:
mac80211: add for 6.6 compatibility

Add 200-Revert-wifi-ath12k-convert-tasklet-to-BH-workqueue-f.patch
for 6.6 compatibility

Signed-off-by: Mantas Pucka <mantas@8devices.com>

mantas-p commented:
Yes I saw that, in my branch I brought back enough patches to compile the driver

namiltd committed:
rtl8812au-ct: update for compatibility with backports 6.14.5

Update for compatibility with backports 6.14.5.
Adding link_id to get_txpower functions parameters
via 110-link_id.patch.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

Noltari commented on package/kernel/mac80211/patches/ath11k/950-wifi-ath11k-secs_to_jiffies-to-msec.patch:
Shouldn't we backport the following commits for this?
https://github.com/torvalds/linux/commit/b35108a51cf7bab58d7eace1267d7965978bcdb8
https://github.com/torvalds/linux/commit/bb2784d9ab49587ba4fbff37a319fff2924db289

Noltari replied:
https://github.com/openwrt/openwrt/pull/18941

Noltari commented on package/kernel/mac80211/patches/brcm/996-brcmfmac-define-devm_clk_get_optional_enabled_with_rate.patch:
Shouldn't we backport https://github.com/torvalds/linux/commit/9934a1bd45b2b03f6d1204a6ae2780d3b009799f for this?

Noltari replied:
https://github.com/openwrt/openwrt/pull/18942

namiltd commented:
The rtw_usb_init_rx function code from backports version 6.14.5 has more differences compared to version 6.12.6. I don't know if it's enough.

Noltari committed:
mac80211: realtek: rtw88: patch for kernels lower than 6.9

Added patch 240-realtek-rtw88-BH-workqueue.patch fixing rtw88
compatibility with kernels lower than 6.9, including 6.6.
These kernels do not support BH workqueues, hence switched
to singlethread workqueue.

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

Gingernut1978 commented:
Nice work

rmandrad commented:
@Ansuel can you help with ath12k ?

Ansuel commented:
What is the problem?

Il Mer 28 Mag 2025, 20:52 Rudy Andram ***@***.***> ha scritto:

> *rmandrad* left a comment (openwrt/openwrt#18911)
> <https://github.com/openwrt/openwrt/pull/18911#issuecomment-2917304796>
>
> @Ansuel <https://github.com/Ansuel> can you help with ath12k ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/18911#issuecomment-2917304796>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE2ZMQWJ5G36TOS74BOS5DD3AYAVDAVCNFSM6AAAAAB53MQMU6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSMJXGMYDINZZGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

namiltd commented:
Some patches for ath12k (removed in https://github.com/openwrt/openwrt/pull/18911/commits/8bb2208559b470bc8f48a60d3761ea5af9156e2b) require an update.

namiltd commented:
@Ansuel Some patches for ath12k (removed in https://github.com/openwrt/openwrt/pull/18911/commits/8bb2208559b470bc8f48a60d3761ea5af9156e2b) require an update.

namiltd commented:
In both cases:
```
pc : cfg80211_chandef_valid+0x660/0x75c [cfg80211]                                                                                                                   
lr : cfg80211_update_last_available+0x50/0xa0 [cfg80211] 
```

namiltd commented:
I fixed 310-cfg80211-allow-grace-period-for-DFS-available-after-.patch. You can test again.

mantas-p commented:
Now it no longer crashes the kernel, but prints a warning with very similar backtrace twice:

```
root@OpenWrt:~# ifconfig phy0-ap0 down             
[  140.775319] ------------[ cut here ]------------           
[  140.775358] WARNING: CPU: 0 PID: 2840 at backports-6.14.5/net/wireless/chan.c:304 cfg80211_chandef_valid+0x6fc/0x71c [cfg80211]
[  140.779019] Modules linked in: pppoe ppp_async nft_fib_inet nf_flow_table_inet ath12k(O) ath11k_ahb(O) ath11k(O) pppox ppp_generic nft_reject_ipv6 nft_reject_ipv4 nft_reject_inet nft_reject nft_redir nft_quota nft_numgen nft_nat nft_masq nft_log nft_limit nft_hash nft_flow_offload nft_fib_ipv6 nft_fib_ipv4 nft_fib nft_ct nft_chain_nat nf_tables nf_nat nf_fc
[  140.838807] CPU: 0 UID: 0 PID: 2840 Comm: ifconfig Tainted: G           O       6.12.30 #0
[  140.861044] Tainted: [O]=OOT_MODULE
[  140.869107] Hardware name: 8devices Mango-DVK (DT)
[  140.872496] pstate: 20400005 (nzCv daif +PAN -UAO -TCO -DIT -SSBS BTYPE=--)
[  140.877360] pc : cfg80211_chandef_valid+0x6fc/0x71c [cfg80211]
[  140.884215] lr : cfg80211_update_last_available+0x50/0xa0 [cfg80211]
[  140.890119] sp : ffffffc086eb3790                                          
[  140.896624] x29: ffffffc086eb3790 x28: ffffff8002d82b80 x27: ffffff80065e83c0
[  140.899843] x26: 0000000000001042 x25: ffffff8004ee4910 x24: 0000000000000001
[  140.906962] x23: ffffff8004ee4000 x22: 00000000000003e8 x21: 0000000000000050
[  140.914080] x20: ffffff80065e83c0 x19: 00000000368a4a80 x18: 0000000000000000
[  140.921198] x17: 0000000000000000 x16: 0000000000000000 x15: 0000007fe49ecd18
[  140.928315] x14: ffffffffffffffff x13: 0000000000000000 x12: 0101010101010101
[  140.935434] x11: 0000000000000040 x10: ffffffc080cc8b48 x9 : 0000000000000000
[  140.942552] x8 : ffffffc086eb3ae0 x7 : 0000000000000000 x6 : 000000000000003f
[  140.949670] x5 : 0000000000000001 x4 : 0000000000000000 x3 : 00000000000003e8
[  140.956787] x2 : 0000000000000050 x1 : 00000000004f7f90 x0 : ffffff80065e83c0
[  140.963906] Call trace:
[  140.971015]  cfg80211_chandef_valid+0x6fc/0x71c [cfg80211]                                                                                                                                                                                                                                                                                                             
[  140.973277]  cfg80211_update_last_available+0x50/0xa0 [cfg80211]
[  140.978833]  cfg80211_leave_mesh+0x158/0x218 [cfg80211]
[  140.984996]  cfg80211_stop_ap+0x58/0xe0 [cfg80211]
[  140.989942]  cfg80211_leave+0x154/0x188 [cfg80211]
[  140.994804]  cfg80211_register_netdevice+0x20c/0x6c4 [cfg80211]
[  140.999580]  raw_notifier_call_chain+0x54/0x80
[  141.005393]  call_netdevice_notifiers_info+0x58/0xa4
[  141.009909]  __dev_close_many+0x50/0x120
[  141.015029]  __dev_change_flags+0xa0/0x1bc
[  141.018934]  dev_change_flags+0x24/0x74
[  141.022840]  devinet_ioctl+0x330/0x5e4
[  141.026573]  inet_ioctl+0x230/0x280
[  141.030392]  sock_ioctl+0x2f4/0x450
[  141.033777]  __arm64_sys_ioctl+0x364/0xe60
[  141.037251]  invoke_syscall.constprop.0+0x5c/0x100
[  141.041419]  do_el0_svc+0x40/0xe0
[  141.046191]  el0_svc+0x24/0xb8
[  141.049575]  el0t_64_sync_handler+0x120/0x12c
[  141.052529]  el0t_64_sync+0x178/0x17c
[  141.056956] ---[ end trace 0000000000000000 ]---
[  141.060759] ------------[ cut here ]------------
```

namiltd commented:
@mantas-p Please check this patch: [310-cfg80211-allow-grace-period-for-DFS-available-after-.patch](https://github.com/user-attachments/files/20518415/310-cfg80211-allow-grace-period-for-DFS-available-after-.patch). Copy it to package/kernel/mac80211/patches/subsys

namiltd commented:
@mantas-p Could you renew the ath12k test? The crash problem is solved now.

mantas-p committed:
mac80211: ath12k: update MU-MIMO and 160MHz support patches

Updated to v14 version from [1], add 003-* series from upstream

[1] https://patchwork.kernel.org/project/linux-wireless/cover/20250521224539.355985-1-quic_pradeepc@quicinc.com/

Signed-off-by: Mantas Pucka <mantas@8devices.com>

mantas-p commented:
@namiltd No more crashes here as well, good job. 

I've updated 104* patch series, please see https://github.com/mantas-p/openwrt/commit/41286d0cfd567e700c723f479fe04f7a057b8133 .
Note that 104-1* and 104-2* are not merged upstream and should not be dropped in `mac80211: drop patches accepted upstream ` commit

namiltd commented:
Ping @hauke

AuthorReflex commented:
Tested on:
mediatek/flogic Cudy TR3000 (6.12)
mediatek/mt7622 Xiaomi AX6s (6.12)
ramips/mt7621 Beeline Smartbox pro, Wifire NBN-S1500 (6.12)
ramips/mt7620 Xiaomi Mi Router 3 (6.6)

danpawlik commented:
@namiltd I tested latest changes and... don't see wireless devices. I will create build from scratch and check again if all is ok.
If someone else have bpi r4 - please make a test. Thanks

danpawlik commented:
@namiltd works fine. Just be14 from time to time require strokes.
In other words, from time to time be14 is like not detected. Need to power off for a while or press the board against the slot even though it is screwed on. Just the last time I bought something from sinovoip :)

LGTM

Ansuel commented:
@danpawlik He(or She???) just wants love <3

namiltd committed:
mac80211: drop patches accepted upstream

Drop patches accepted upstream

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
mac80211: manually refresh patches

Manually refresh patches

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
mac80211: rtw88: change invalid patch name

Rename 091-v6.15wifi-rtw88-Enable-the-new-RTL8814AE-RTL8814AU-driver.patch
to 092-v6.15-wifi-rtw88-Enable-the-new-RTL8814AE-RTL8814AU-driver.patch.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd commented:
Updated with recent changes in rtw88 in main branch.

namiltd commented:
Ping @mj22226. Please look at this PR. Shouldn't https://github.com/torvalds/linux/commit/13221be72034d1c34630ab124c43438aefe7e656 and https://github.com/torvalds/linux/commit/3e3aa566dd1803f1697530de6c8489a8350765b3 that you mentioned in https://github.com/openwrt/openwrt/commit/4a1aca53761643404a81f8023eb16b53142423df also be added?

namiltd commented:
So it looks like it's ok.

mj22226 commented:
> So it looks like it's ok.

@namiltd yes, tested working with rtw88_8821cu, rtw88_8812au, rtw88_8812au usb wifi adapters.

namiltd commented:
@Gingernut1978 Try this patch: [999-fix.patch](https://github.com/user-attachments/files/20717412/999-fix.patch)
Copy it to folder: package/kernel/mac80211/patches/subsys

Gingernut1978 commented:
Applying your patch on top of this PR resolved the issue.

Amazing work.

namiltd committed:
mac80211: convert short fops to debugfs files

Reverse https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=31cb94f71c1ba126b9c684c5882cbf5032d0d1bc
for compatibility with kernel 6.6 and 6.12.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd committed:
mac80211: automatically refresh patches

Automatically refresh patches

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd commented:
Please do not create new accounts, only developer approval has power anyway!

namiltd commented:
@rmandrad. The patch still exists: https://github.com/openwrt/openwrt/pull/18911/commits/6338d31aacba49956b0501481736af675a139664. I didn't make any changes other than a rebase.
We did not find any of the problems you mentioned. Attach the end of the compilation log.

rmandrad commented:
i understand now what happened ... my bad... i just used the mac80211 changes not mt76

neheb commented:
https://git.kernel.org/pub/scm/linux/kernel/git/ath/ath.git/log/drivers/net/wireless/ath/ath9k patch 552 got upstreamed. Might be worth replacing it. Upstream version is slightly different.

namiltd commented:
I am not able to test these changes so I suggest doing it in another PR.

namiltd committed:
mac80211: ath9k: update patch 552

Update patch 552-ath9k-ahb_of.patch to upstream version.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>

namiltd commented:
Patch 552-ath9k-ahb_of.patch, please test it on real hardware.

neheb commented:
Wonderful.

danpawlik commented:
May I ask, if such stack trace is related to this patch https://gist.github.com/danpawlik/bcf2ec89db2a22cf35d1896f63926b51 or something related to mt76 ?

neheb commented:
that looks specific to mt76

namiltd commented:
For the PR to be approved, do I need to wait until the migration to the 6.12 kernel is complete, or do I need to wait for approval from @nbd168 ?

nbd168 commented:
I've pushed the mac80211 update to my staging tree at https://git.openwrt.org/?p=openwrt/staging/nbd.git;a=summary
Unless I see any issues with it, I will push it to main soon. Please test, if you can.

namiltd commented:
@nbd168 I have updated to version 6.14.11 in parallel and the only differences between your result and mine are:

```
--- 552-ath9k-ahb_of.patch	2025-07-04 22:26:15.360185300 +0200
+++ 552-ath9k-ahb_of.patch_	2025-07-04 22:16:26.000000000 +0200
@@ -84,7 +84,7 @@
  	if (ret) {
  		dev_err(&pdev->dev, "failed to initialize device\n");
  		goto err_irq;
-@@ -167,8 +147,8 @@ static struct platform_driver ath_ahb_dr
+@@ -167,11 +147,11 @@ static struct platform_driver ath_ahb_dr
  	
  	.driver = {
  		.name = "ath9k",
@@ -93,4 +93,8 @@
 -	.id_table = ath9k_platform_id_table,
  };
  
- MODULE_DEVICE_TABLE(platform, ath9k_platform_id_table);
+-MODULE_DEVICE_TABLE(platform, ath9k_platform_id_table);
++MODULE_DEVICE_TABLE(of, ath9k_of_match_table);
+ 
+ int ath_ahb_init(void)
+ {
```
Please check if it should be `MODULE_DEVICE_TABLE(platform, ath9k_platform_id_table);` or `MODULE_DEVICE_TABLE(of, ath9k_of_match_table);`?

nbd168 commented:
@namiltd you're right. That chunk accidentally got lost on my end when I manually fixed rejects during rebase. Fixed now, thanks.

nbd168 commented:
Applied, thanks

danpawlik commented:
@nbd168 thanks for quick fix.
After doing test, 5GHz network is available now, but still can not connect to 5GHz or 6GHz network.

nbd168 commented:
@danpawlik does the main branch behave in the same way?

nbd168 commented:
@danpawlik thanks for tracking it down. I will look into it later or tomorrow

nbd168 commented:
@namiltd done, thanks.

danpawlik commented:
> @danpawlik in the mean time, could you please make some debug logs for me with connection attempts on the 2.4 GHz network, and also of the setup during the 'wifi' call? Also, which wpad/hostapd variant are you using?

sure, will do soon (or after walk).
```
root@BPI-R4:~# apk list -I | sort | grep -E 'wpad|hostapd'
hostapd-common-2025.05.23~4b8ac10c-r2 aarch64_cortex-a53 {feeds/base/network/services/hostapd} (BSD-3-Clause) [installed]
wpad-openssl-2025.05.23~4b8ac10c-r2 aarch64_cortex-a53 {feeds/base/network/services/hostapd} (BSD-3-Clause) [installed]
```

grosjo commented:
This is the error I get for ath9k since the [2ac791e]

```
Sun Jul  6 17:23:08 2025 daemon.notice netifd: radio0 (2143): WARNING: Variable 'data' does not exist or is not an array/object                                             
Sun Jul  6 17:23:08 2025 daemon.notice netifd: radio0 (2143): Phy not found                                                                                                 
Sun Jul  6 17:23:08 2025 daemon.notice netifd: radio0 (2143): Could not find PHY for device 'radio0'                                                                        
Sun Jul  6 17:23:08 2025 daemon.notice netifd: Wireless device 'radio0' set retry=0                                                                                         
Sun Jul  6 17:23:08 2025 daemon.crit netifd: Wireless device 'radio0' setup failed, retry=0                                                                                 
Sun Jul  6 17:23:09 2025 daemon.notice netifd: radio0 (2198): WARNING: Variable 'data' does not exist or is not an array/object                                             
Sun Jul  6 17:23:09 2025 daemon.notice netifd: radio0 (2198): Bug: PHY is undefined for device 'radio0'                                                                     
Sun Jul  6 17:23:09 2025 daemon.notice netifd: Wireless device 'radio0' is now down                                                                                         

```

danpawlik commented:
We should move conversation to https://github.com/openwrt/openwrt/issues/19312
